src/app/plugins/crumbl.plugin.ts

```diff
@@ -0,0 +1,134 @@
+// Original bot code: https://github.com/rob-3/crumblbot
+
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { IContainer, IHttpResponse, IMessage, ChannelType, Maybe } from '../../common/types';
+import { MessageEmbed } from 'discord.js';
+
+export class CrumblPlugin extends Plugin {
+  public name: string = 'Crumbl Cookies Plugin';
+  public description: string = 'Returns the cookies available that week at Crumbl';
+  public usage: string = 'crumbl';
+  public pluginAlias = [];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Food;
+  public buildId: string = '';
+  private _lastPost: Maybe<IMessage> = null;
+  
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public _matchId (data: string, regex: RegExp): string
+  {
+    const matches = data.match(regex);
+
+    if (matches !== null && matches.length > 1)
+    {
+      return matches[1];
+    }
+    
+    return '';
+  }
+
+  private _createEmbed(cookies: any, message: IMessage): MessageEmbed[]
+  {
+    return  cookies
+      .filter (
+        (c: any) => c.name !== 'Milk Chocolate Chip' && c.name !== 'Chilled Sugar'
+      )
+      .map((cookie: any, i: number) => {
+          const embed = new MessageEmbed()
+          .setTitle(cookie.name)
+          .setDescription(cookie.description)
+          .setThumbnail(cookie.image);
+          return embed;
+        });
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const regex: RegExp = /"buildId":"(.*?)"/;
+
+    await this.container.httpService
+      .get('https://crumblcookies.com')
+      .then((response: IHttpResponse) => {
+        
+        const data: string = response.data;
+        this.buildId = this._matchId(data, regex); 
+      })
+      .catch((err) => this.container.loggerService.warn(err));      
+
+    await this.container.httpService
+      .get(`https://crumblcookies.com/_next/data/${this.buildId}/index.json`)
+        .then(async (blob: IHttpResponse) => {
+          if (this.buildId === '')
+          {
+            console.log('The build id is empty');
```

Comment:
prefer `this.container.loggerService.warn(...)` if its useful to know.
Also when we catch errors, its best to let the user know something happened, at least give them a response